connection.h

```diff
@@ -31,13 +31,15 @@ namespace QMatrixClient
     class ConnectionData;
 
     class SyncJob;
+    class SyncData;
     class RoomMessagesJob;
     class PostReceiptJob;
     class MediaThumbnailJob;
     class JoinRoomJob;
 
     class Connection: public QObject {
             Q_OBJECT
+            Q_PROPERTY(QString stateSaveFile READ getStateSaveFile WRITE setStateSaveFile)
```

Comment:
If we use `Settings`, as I noted above, this property becomes unnecessary. Any wanting code (including QML) could just dive into the respective settings value.